imap_processing/hit/l2/hit_l2.py

```diff
@@ -0,0 +1,163 @@
+"""IMAP-HIT L2 data processing."""
+
+import logging
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from imap_processing import imap_module_directory
+from imap_processing.hit.hit_utils import get_attribute_manager
+
+logger = logging.getLogger(__name__)
+
+# TODO review logging levels to use (debug vs. info)
+
+
+def hit_l2(dependency: xr.Dataset, data_version: str) -> list[xr.Dataset]:
+    """
+    Will process HIT data to L2.
+
+    Processes dependencies needed to create L2 data products.
+
+    Parameters
+    ----------
+    dependency : xr.Dataset
+        L1B xarray science dataset that is either summed rates
+        standard rates or sector rates.
+    data_version : str
+        Version of the data product being created.
+
+    Returns
+    -------
+    processed_data : list[xarray.Dataset]
+        List of L2 dataset.
+    """
+    logger.info("Creating HIT L2 science datasets")
+    # Create the attribute manager for this data level
+    attr_mgr = get_attribute_manager(data_version, "l2")
+
+    # Logical sources for the three L2 science products.
+    # TODO: add logical sources for other l2 products once processing functions
+    #  are written. "imap_hit_l2_standard-fluxes", "imap_hit_l2_sectored-fluxes"
+    logical_sources = ["imap_hit_l2_summed-fluxes"]
+
+    # TODO: Write functions to create the following datasets
+    #  Process sectored rates dataset
+    #  Process standard rates dataset
+
+    # Create L2 datasets
+    l2_datasets: list = []
+
+    if "imap_hit_l1b_summed-rates" in dependency.attrs["Logical_source"]:
+        # Process science data to L2 datasets
+        l2_datasets.append(process_summed_flux_data(dependency))
+        logger.info("HIT L2 summed flux dataset created")
+
+    # Update attributes and dimensions
+    for dataset, logical_source in zip(l2_datasets, logical_sources):
+        dataset.attrs = attr_mgr.get_global_attributes(logical_source)
+
+        # TODO: Add CDF attributes to yaml once they're defined for L2 science data
+        #  consider moving attribute handling to hit_utils.py
+        # Assign attributes and dimensions to each data array in the Dataset
+        for field in dataset.data_vars.keys():
+            try:
+                # Create a dict of dimensions using the DEPEND_I keys in the
+                # attributes
+                dims = {
+                    key: value
+                    for key, value in attr_mgr.get_variable_attributes(field).items()
+                    if "DEPEND" in key
+                }
+                dataset[field].attrs = attr_mgr.get_variable_attributes(field)
+                dataset[field].assign_coords(dims)
+            except KeyError:
+                print(f"Field {field} not found in attribute manager.")
+                logger.warning(f"Field {field} not found in attribute manager.")
```

Comment:
It is OK to continue processing if this happens or do you want to re-raise?

You probably only need one or the other of the print / logger.warning